swift/ql/lib/codeql/swift/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -1039,20 +1105,65 @@ private class OptionalSomeContentSet extends ContentSet {
   }
 }
 
-private newtype TDataFlowType = TODO_DataFlowType()
+newtype TDataFlowType =
+  TDataFlowTypeType(Type t) {
+    t.getCanonicalType() = t and
+    not t instanceof AnyFunctionType
+  } or
+  TTopFunctionType() or
+  TDataFlowFunctionType(DataFlowCallable c)
```

Comment:
What's the difference between `TTopFunctionType` and `TDataFlowFunctionType`?  Why do we need both?

Reply:
TTopFunctionType allows any function type, DataFlowFunctionType is a specific callable (and can be used to improve precision of lambda call resolution)